scripts/nmt/nmt.py

```diff
@@ -0,0 +1,125 @@
+
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+
+import grpc
+import riva.client.proto.riva_nmt_pb2 as riva_nmt
+import riva.client.proto.riva_nmt_pb2_grpc as riva_nmt_srv
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description="Streaming transcription via Riva AI Services")
+    parser.add_argument("--riva_uri", default="localhost:50051", type=str, help="URI to GRPC server endpoint")
+    parser.add_argument(
+        "--text", default="mir Das ist mir Wurs, bien ich ein berliner", type=str, help="Text to translate"
+    )
+    parser.add_argument("--model_name", default="riva-nmt", type=str, help="model to use to translate")
+    parser.add_argument(
+        "--src_language", default="en", type=str, help="Source language (according to BCP-47 standard)"
+    )
+    parser.add_argument(
+        "--tgt_language", default="de", type=str, help="Target language (according to BCP-47 standard)"
+    )
+    parser.add_argument("--ssl_cert", type=str, default="", help="Path to SSL client certificatates file")
+    parser.add_argument("--text_file", type=str, help="Path to file for translation")
+    parser.add_argument("--batch_size", type=int, default=8, help="Batch size to use for file translation")
+    parser.add_argument("--list_models", default=False, action='store_true', help="List available models")
+    parser.add_argument(
+        "--use_ssl", default=False, action='store_true', help="Boolean to control if SSL/TLS encryption should be used"
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    def request(inputs):
+        nmt_client = riva_nmt_srv.RivaTranslationStub(channel)
+        nmt_request = riva_nmt.TranslateTextRequest()
+        nmt_request.texts.extend(inputs)
+        nmt_request.model = args.model_name
+        nmt_request.source_language = args.src_language
+        nmt_request.target_language = args.tgt_language
+        try:
+            response = nmt_client.TranslateText(nmt_request)
+            for translation in response.translations:
+                print(translation.text)
+        except grpc.RpcError as e:
+            if e.code() == grpc.StatusCode.INVALID_ARGUMENT:
+                result = {'msg': 'invalid arg error'}
+            elif e.code() == grpc.StatusCode.ALREADY_EXISTS:
+                result = {'msg': 'already exists error'}
+            elif e.code() == grpc.StatusCode.UNAVAILABLE:
+                result = {'msg': 'server unavailable check network'}
+            print(f"{result['msg']} : {e.details()}")
+
+    args = get_args()
+
+    if args.ssl_cert != "" or args.use_ssl:
+        root_certificates = None
+        if args.ssl_cert != "" and os.path.exists(args.ssl_cert):
+            with open(args.ssl_cert, 'rb') as f:
+                root_certificates = f.read()
+        creds = grpc.ssl_channel_credentials(root_certificates)
+        channel = grpc.secure_channel(args.riva_uri, creds)
```

Comment:
For establishing connection we use `Auth` [class](class Auth:). It includes creating a channel and loading credentials from `ssl_cert`. You may see an example of `Auth` usage here https://github.com/nvidia-riva/python-clients/blob/ad71e0e766d6eb2a2b4798adeee2584f0c126c51/scripts/asr/transcribe_file.py#L69.

Reply:
added